src/test/java/core/basesyntax/service/impl/WriterServiceImplTest.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.service.WriterService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class WriterServiceImplTest {
+    private static WriterService writerService;
+
+    @BeforeAll
+    static void beforeAll() {
+        writerService = new WriterServiceImpl();
+    }
+
+    @Test
+    void writeToFile_validPath_ok() {
+        String report = "fruit,quantity" + System.lineSeparator()
+                + "banana,152" + System.lineSeparator()
+                + "apple,90" + System.lineSeparator();
+        String validFilePath = "test_valid_report.csv";
+        writerService.writeToFile(validFilePath,report);
+        Path path = Paths.get(validFilePath);
+        List<String> strings = null;
+        try (Stream<String> lines = Files.lines(path)) {
+            strings = lines.collect(Collectors.toList());
+        } catch (IOException e) {
+            fail("Exсeption shouldn't be thrown in this test");
+        }
+        String[] transactions = report.split(System.lineSeparator());
+        for (int i = 0; i < strings.size(); i++) {
+            String expected = strings.get(i);
+            assertEquals(expected,transactions[i],
+                    "transactions differ at element: " + i);
+        }
+    }
```

Comment:
Looks too overcomplicated. Consider making it prettier with `assertIterableEquals` and some some values to constant separation.